### DISCUSSION TITLE
Azure Container App Registries-Adding Azure Container Registry from other subscription not working

### DISCUSSION THREAD

**Discussion by ranjeetkpGit** [pulumi/pulumi-azure-native#1970] [Q&A]
### Discussed in https://github.com/pulumi/pulumi/discussions/10680

<div type='discussions-op-text'>

<sup>Originally posted by **ranjeetkpGit** September 12, 2022</sup>

</div>

Hi,
I am creating an azure container app with Pulumi native v1.66.
My container app in azure 'subscription1' and azure container registry is in 'subscription2'.
I am trying to add Registry when creating container app but it throws error
ERROR: The resource with name 'testacr' and type 'Microsoft.ContainerRegistry/registries' could not be found in subscription 'subscription1'.

```
var containerApp= new ContainerApp(appName, new ContainerAppArgs
            {
                Configuration = new ConfigurationArgs
                {
                    Dapr = new DaprArgs
                   {
                     AppPort = 80,
                     AppProtocol = "http",
                     Enabled = true,
                   },
                    Ingress = new IngressArgs
                    {
                        External = false,
                        TargetPort = 80
                    },
                    Secrets = new InputList<SecretArgs>
                    {
                        new SecretArgs { Name = "registry-password", Value = "testpassword"}
                    },
                    Registries = new InputList<RegistryCredentialsArgs>   // From other subscription
                    {
                        new RegistryCredentialsArgs
                        {
                            Server ="[testacr.azurecr.io](http://testacr.azurecr.io/)",
                            Username = "testacr",
                            PasswordSecretRef = "registry-password"
                        }
                    }
                }
}
```

**Comment by danielrbradley** [comment]
Hi @ranjeetkpGit I'd suggest trying two options:

1. Try specifying an `identity` in the `RegistryCredentials`.
2. If this is something you can do via the console, use the wizard to export an ARM template and inspect the values Azure would set to make this work. The Pulumi provider is exactly equivelent in functionality to ARM templates.

**Reply by ranjeetkpGit** [reply]
Hi @danielrbradley,
Thanks for reply,

The container registry in different subscription is an existing resource and created by someone else

In workaround I have tried following CLI command to add registry from other subscription to container app and it works

az containerapp registry set -n MyContainerapp -g MyResourceGroup
--server MyExistingContainerappRegistry.azurecr.io --username MyRegistryUsername --password MyRegistryPassword

Do we have any similar workaround in Pulumi?
